plasma_cash/client/history.py

```diff
@@ -0,0 +1,37 @@
+import rlp
+from rlp.sedes import big_endian_int, binary, CountableList
+
+from .exceptions import TxHistoryNotFoundException
+
+
+class History(rlp.Serializable):
+    EMPTY_TX = b'\x00' * 32
+
+    fields = [
+        ('deposit_tx_blk_num', big_endian_int),
+        ('latest_tx_blk_num', big_endian_int),
+        ('latest_blk_num', big_endian_int),
+        ('tx_history', CountableList(binary)),
+        ('proofs', CountableList(binary))
+    ]
+
+    def __init__(self, deposit_tx_blk_num, tx, proof):
+        self.deposit_tx_blk_num = deposit_tx_blk_num
+        self.latest_tx_blk_num = deposit_tx_blk_num
+        self.latest_blk_num = deposit_tx_blk_num
+        self.tx_history = [tx]
+        self.proofs = [proof]
+
+    def update_tx_history(self, tx, proof):
+        # TODO: should add re-org protection
+        self.latest_blk_num += 1
+        if tx != self.EMPTY_TX:
+            self.latest_tx_blk_num = self.latest_blk_num
```

Comment:
I think we can get rid of this `latest_blk_num`. We can calculate it by `len(tx_history) + self.deposit_tx_blk_num`. However, if we are using the `blk_num_list` as I suggested above, we don't need to count num anymore.